code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -25,6 +25,7 @@
 	var/addiction_threshold = 0
 	var/addiction_stage = 0
 	var/overdosed = 0 // You fucked up and this is now triggering its overdose effects, purge that shit quick.
+	var/stun_timer = 0 // How many ticks since you last resisted a stun, tracked by stun-resisting reagents.
```

Comment:
Could also add the threshold to when stuns can be resisted again, E.g morphine's would be stun_threshold=4